presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -316,8 +316,15 @@ public PlanOptimizers(
                                 new TransformCorrelatedInPredicateToJoin(), // must be run after PruneUnreferencedOutputs
                                 new TransformCorrelatedScalarSubquery(), // must be run after TransformCorrelatedScalarAggregationToJoin
                                 new TransformCorrelatedLateralJoinToJoin(),
-                                new ImplementFilteredAggregations())),
-                new TransformCorrelatedSingleRowSubqueryToProject(),
+                                new ImplementFilteredAggregations(),
+                                new InlineProjections(),
+                                new RemoveRedundantIdentityProjections())),
+                new IterativeOptimizer(
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.of(
+                                new TransformCorrelatedSingleRowSubqueryToProject())),
```

Comment:
can you run projection related rules within this iterative optimizer?

Reply:
Running them in same iterative optimizer runs this rule before InlineProjections, which leads to incorrect results, even if I add InlineProjections before the new rule in the same optimizer.

Reply:
Why is so? Rule should not rewrite is is not able to rewrite.

Reply:
I am not sure why. InlineProjections does not fire before TransformCorrelatedSingleRowSubqueryToProjection even if it is added before the new rule in the optimizer

Reply:
Iterative optimizer keeps to fire its rules until none of the changed the plan. So `InlineProjections` does not need be fired before `TransformCorrelatedSingleRowSubqueryToProjection`, it could be run after and after that `TransformCorrelatedSingleRowSubqueryToProjection` will get another chance.